### PR TITLE
add modern health check banner + expiration tracking

### PR DESCRIPTION
**Pull request by pablodanswer**
re-implemented from EE

### CODE REVIEW

File: deployment/docker_compose/docker-compose.dev.yml [hunk 1]
```diff
@@ -231,6 +231,9 @@ services:
 
       # Enterprise Edition only
       - ENABLE_PAID_ENTERPRISE_EDITION_FEATURES=${ENABLE_PAID_ENTERPRISE_EDITION_FEATURES:-false}
+    tty: true
```
Comment (Weves) [inline deployment/docker_compose/docker-compose.dev.yml]: what do these do? can we add a comment here?

File: backend/danswer/server/manage/users.py [hunk 2]
```diff
@@ -246,9 +250,33 @@ async def get_user_role(user: User = Depends(current_user)) -> UserRoleResponse:
...
+            return None, None
+
+    except Exception as e:
+        logger.error(f"Error fetching AccessToken: {e}")
+        return None, None
+
+
 @router.get("/me")
-def verify_user_logged_in(
+async def verify_user_logged_in(
```
Comment (Weves) [inline backend/danswer/server/manage/users.py]: hmm, why does this need to be async now?

File: backend/danswer/server/manage/users.py [hunk 3]
```diff
@@ -246,9 +250,33 @@ async def get_user_role(user: User = Depends(current_user)) -> UserRoleResponse:
     return UserRoleResponse(role=user.role)
 
 
+def get_current_token_info(user: User | None, db_session: Session):
+    if not isinstance(user, User):
```
Comment (Weves) [inline backend/danswer/server/manage/users.py]: prefer `if user is None`

File: web/src/app/chat/ChatPage.tsx [hunk 4]
```diff
@@ -1072,6 +1072,7 @@ export function ChatPage({
   const [editingRetrievalEnabled, setEditingRetrievalEnabled] = useState(false);
   const sidebarElementRef = useRef<HTMLDivElement>(null);
   const innerSidebarElementRef = useRef<HTMLDivElement>(null);
+  // const secondsUntilExpiration = getSecondsUntilExpiration(user);
```
Comment (Weves) [inline web/src/app/chat/ChatPage.tsx]: nit: remove

File: web/src/app/chat/ChatPage.tsx [hunk 5]
```diff
@@ -1088,6 +1089,9 @@ export function ChatPage({
     <>
       <HealthCheckBanner />
       <InstantSSRAutoRefresh />
+      {/* <div className="m-3">
```
Comment (Weves) [inline web/src/app/chat/ChatPage.tsx]: nit: remove

File: web/src/lib/types.ts [hunk 6]
```diff
@@ -19,6 +19,9 @@ export interface User {
   role: "basic" | "admin";
   preferences: UserPreferences;
   status: UserStatus;
+  current_token_created_at: Date;
```
Comment (Weves) [inline web/src/lib/types.ts]: all these can be `null`, right?

File: backend/danswer/db/models.py [hunk 7]
```diff
@@ -138,7 +143,7 @@ class User(SQLAlchemyBaseUserTableUUID, Base):
 
 
 class AccessToken(SQLAlchemyBaseAccessTokenTableUUID, Base):
-    pass
+    expiry_length: Mapped[int] = mapped_column(Integer, nullable=False)
```
Comment (Weves) [inline backend/danswer/db/models.py]: do we need to add this? Can we instead compute it on the fly in the `/me` endpoint?

**Review by Weves — CHANGES_REQUESTED** [review]
Overall, looks great!

Small thing, I'm seeing this:

Screenshot 2024-07-10 at 6 37 11 PM

File: backend/danswer/db/models.py (hunk 7 above)
Comment (pablonyx) [reply backend/danswer/db/models.py]: Certainly can treat the expiry length time in seconds (environment variable) as a constant and use that instead

**Comment by pablodanswer** [comment]
Should be set now!

**Pull request merged**